rewrite-maven/src/main/java/org/openrewrite/maven/ChangePropertyKey.java

```diff
@@ -0,0 +1,99 @@
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.marker.SearchResult;
+import org.openrewrite.maven.tree.Dependency;
+import org.openrewrite.maven.tree.ResolvedManagedDependency;
+import org.openrewrite.maven.tree.ResolvedPom;
+import org.openrewrite.xml.ChangeTagNameVisitor;
+import org.openrewrite.xml.ChangeTagValueVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class ChangePropertyKey extends Recipe {
+
+    @Option(displayName = "Old key",
+            description = "The old name of the property key to be replaced.",
+            example = "junit.version")
+    String oldKey;
+
+    @Option(displayName = "New key",
+            description = "The new property name to use.",
+            example = "version.org.junit")
+    String newKey;
+
+    @Override
+    public String getDisplayName() {
+        return "Change Maven project property key";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Changes the specified Maven project property key leaving the value intact.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new MavenVisitor<ExecutionContext>() {
+            @Override
+            public Xml visitDocument(Xml.Document document, ExecutionContext executionContext) {
+                String oldKeyAsProperty = "${" + oldKey + "}";
+                ResolvedPom pom = getResolutionResult().getPom();
+                if (pom.getProperties().containsKey(oldKey)) {
+                    return SearchResult.found(document);
+                }
+                for (Dependency dependency : pom.getRequestedDependencies()) {
+                    if (dependency.getVersion().contains(oldKeyAsProperty)) {
+                        return SearchResult.found(document);
+                    }
+                }
+                for (ResolvedManagedDependency dependency : pom.getDependencyManagement()) {
+                    if (dependency.getVersion().contains(oldKeyAsProperty)) {
+                        return SearchResult.found(document);
+                    }
+                }
+                return document;
+            }
+        };
+    }
+
+    @Override
+    public TreeVisitor<Xml, ExecutionContext> getVisitor() {
+        return new MavenIsoVisitor<ExecutionContext>() {
+            final String oldKeyAsProperty = "${" + oldKey + "}";
+            final String newKeyAsProperty = "${" + newKey + "}";
+
+            @Override
+            public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                if (isPropertyTag() && oldKey.equals(tag.getName())) {
+                    doAfterVisit(new ChangeTagNameVisitor<>(tag, newKey));
+                }
+                if (oldKeyAsProperty.equals(tag.getValue().orElse(null))) {
+                    doAfterVisit(new ChangeTagValueVisitor<>(tag, newKeyAsProperty));
```

Comment:
Do you think we should support cases where the property name is only part of the tag value?
So for instance `<value>http://some/doc/url/${guava.version}</value>`

Reply:
Good idea.
I would ignore elements that have elements, as `<value>${guava.version}<foo></foo></value>` seems impossible to me in typical maven land. But we can do a contains() check and replace it accordingly.

Reply:
Done.